File: src/main/java/kitchenpos/menu/tobe/domain/menu/MenuRepository.java
package kitchenpos.menu.tobe.domain.menu;

import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface MenuRepository {
    Menu save(Menu menu);

    Optional<Menu> findById(UUID id);

    List<Menu> findAll();

    List<Menu> findAllByIdIn(List<UUID> ids);

    List<Menu> findAllByProductId(UUID productId);
}

File: src/main/kotlin/kitchenpos/menu/tobe/domain/MenuRepository.kt
package kitchenpos.menu.tobe.domain

import java.util.*
import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Repository

@Repository("tobeMenuRepository")
@Primary
interface MenuRepository {
    fun save(menu: Menu): Menu
    fun findById(id: UUID): Optional<Menu>
    fun findAll(): List<Menu>
    fun findAllByIdIn(ids: List<UUID>): List<Menu>
    fun findAllByProductId(productId: UUID): List<Menu>
}
